apiserver/billing/odoostorage/odoo/odoo16/odoo16.go

```diff
@@ -0,0 +1,451 @@
+package odoo16
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	billingv1 "github.com/appuio/control-api/apis/billing/v1"
+	"github.com/appuio/control-api/apiserver/billing/odoostorage/odoo"
+
+	odooclient "github.com/appuio/go-odoo"
+)
+
+const VSHNAccountingContactNameKey = "billing.appuio.io/vshn-accounting-contact-name"
+
+// Used to identify the accounting contact of a company.
+const roleAccountCategory = 70
+const companyCategory = 2
+const invoiceType = "invoice"
+
+// Used to generate the UUID for the .metadata.uid field.
+var metaUIDNamespace = uuid.MustParse("51887759-C769-4829-9910-BB9D5F92767D")
+
+var roleAccountFilter = odooclient.NewCriterion("category_id", "in", []int{roleAccountCategory})
+var activeFilter = odooclient.NewCriterion("active", "=", true)
+var notInflightFilter = odooclient.NewCriterion("vshn_control_api_inflight", "=", false)
+var mustInflightFilter = odooclient.NewCriterion("vshn_control_api_inflight", "!=", false)
+
+var fetchPartnerFieldOpts = odooclient.NewOptions().FetchFields(
+	"id",
+	"type",
+	"name",
+	"display_name",
+	"country_id",
+	"commercial_partner_id",
+	"contact_address",
+
+	"child_ids",
+	"user_ids",
+
+	"email",
+	"phone",
+	"street",
+	"street2",
+	"city",
+	"zip",
+	"country_id",
+
+	"parent_id",
+	"vshn_control_api_meta_status",
+	"vshn_control_api_inflight",
+	"x_invoice_contact",
```

Comment:
This might be a copy-pasta leftover?

Reply:
Sorry i specifically meant `"x_invoice_contact"`. `x_` is not allowed anymore the last time i spoke to tobru?

Reply:
Currently, in the test odoo16, that field is still called the same (with the x prefix). I'll check back with tobru.

Reply:
intermediate result: nobody really knows what the field does and everyone's generally confused. 🙃

Reply:
final result: the field is dropped, the data will go somewhere else, and therefore the datamodel will change and I have to update the implementation.